src/components/LendModals/index.tsx

```diff
@@ -182,36 +216,45 @@ export const QuickModalContent: React.FC<QuickModalContentProps> = ({
       setUpdatedAsset(undefined);
       if (modalType === 'borrow' || modalType === 'repay') {
         const updatedBorrowBalance =
-          convertBNToNumber(asset.borrowBalance, asset.underlyingDecimals) +
+          convertBNToNumber(
+            currentAsset.borrowBalance,
+            currentAsset.underlyingDecimals,
+          ) +
           (modalType === 'repay' ? -1 : 1) * numValue;
         const updatedTotalBorrowNum =
-          convertBNToNumber(asset.totalBorrow, asset.underlyingDecimals) +
+          convertBNToNumber(
+            currentAsset.totalBorrow,
+            currentAsset.underlyingDecimals,
+          ) +
           (modalType === 'repay' ? -1 : 1) * numValue;
-        const web3 = asset.cToken.sdk.web3;
+        const web3 = currentAsset.cToken.sdk.web3;
         const updatedTotalBorrow = web3.utils.toBN(
           (
             updatedTotalBorrowNum *
-            10 ** Number(asset.underlyingDecimals)
+            10 ** Number(currentAsset.underlyingDecimals)

```

Comment:
Will this work we're doing the multiplication before converting to Big number? Shouldn't we be using big number to do the multiplication instead?